src/Plugins/ComplexCore/test/RawBinaryReaderTest.cpp

```diff
@@ -0,0 +1,446 @@
+/** Testing Notes:
+ *
+ *  Case1: This tests when skipHeaderBytes equals 0, and checks to see if the data read is the same as the data written.
+ *
+ *  Case2: This tests when the wrong scalar type is selected. (The total number of bytes in the file does not evenly divide by the scalar type size).
+ *
+ *  Case3: This tests when the wrong component size is chosen. (The total number of scalar elements in the file does not evenly divide by the chosen component size).
+ *
+ *  Case4: This tests when skipHeaderBytes is non-zero, and checks to see if the data read is the same as the data written.
+ *
+ *  Case5: This tests when skipHeaderBytes equals the file size
+ */
+
+/** we are going to use a fairly large array size because we want to exercise the
+ * possibilty that we can not write the data or read the data from the file in one
+ * step in the filter
+ */
+
+#include <catch2/catch.hpp>
+
+#include "complex/Common/ScopeGuard.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataStore.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ChoicesParameter.hpp"
+#include "complex/Parameters/FileSystemPathParameter.hpp"
+#include "complex/Parameters/NumberParameter.hpp"
+#include "complex/Parameters/NumericTypeParameter.hpp"
+
+#include <filesystem>
+namespace fs = std::filesystem;
+
+#include "ComplexCore/Filters/RawBinaryReaderFilter.hpp"
+#include "complex/unit_test/complex_test_dirs.hpp"
+
+using namespace complex;
+
+const std::string k_TestOutput = (fs::path(complex::unit_test::k_BinaryDir.str()).append("RawBinaryReaderTest").append("Output.bin")).string();
+const DataPath k_CreatedArrayPath = DataPath(std::vector<std::string>{"Test_Array"});
+
+constexpr int32_t k_RbrNumComponentsError = -392;
+constexpr int32_t k_RbrWrongType = -393;
+constexpr int32_t k_RbrSkippedTooMuch = -395;
```

Comment:
```suggestion
namespace
{
const std::string k_TestOutput = (fs::path(complex::unit_test::k_BinaryDir.str()).append("RawBinaryReaderTest").append("Output.bin")).string();
const DataPath k_CreatedArrayPath = DataPath(std::vector<std::string>{"Test_Array"});

constexpr int32 k_RbrNumComponentsError = -392;
constexpr int32 k_RbrWrongType = -393;
constexpr int32 k_RbrSkippedTooMuch = -395;
} // namespace
```
Should be in an anonymous namespace.